### PR TITLE
Configure cloud from keyvault uri

### PR DESCRIPTION
**Pull request by yiliuTo**
To support keyvault jca in different Azure cloud environments, this pr configures keyvault resource uri and aad authentication automatically according to the property of **azure.keyvault.uri** provided by customers

### CODE REVIEW

**Comment by chenrujun** [comment]
Hi, @yiliuTo .
Please fix [the pipeline failure](https://dev.azure.com/azure-sdk/public/_build/results?buildId=842102&view=logs&j=b70e5e73-bbb6-5567-0939-8415943fadb9&t=ba3cf43a-b2bf-50f9-f95f-93049ca44ad1&l=604). 🙏

File: sdk/keyvault/azure-security-keyvault-jca/src/main/java/com/azure/security/keyvault/jca/Constants.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+    public static final String KEY_VAULT_BASE_URI_GLOBAL = "https://vault.azure.net";
+    public static final String KEY_VAULT_BASE_URI_CN = "https://vault.azure.cn";
+    public static final String KEY_VAULT_BASE_URI_US = "https://vault.usgovcloudapi.net";
+    public static final String KEY_VAULT_BASE_URI_DE = "https://vault.microsoftazure.de";
+
+    public static final String AAD_LOGIN_GLOBAL_URI = "https://login.microsoftonline.com/";
+    public static final String AAD_LOGIN_CN_URI = "https://login.partner.microsoftonline.cn/";
+    public static final String AAD_LOGIN_US_URI = "https://login.microsoftonline.us/";
+    public static final String AAD_LOGIN_DE_URI = "https://login.microsoftonline.de/";
```
Comment (chenrujun) [inline sdk/keyvault/azure-security-keyvault-jca/src/main/java/com/azure/security/keyvault/jca/Constants.java:19]: To keep the same style, use `AAD_LOGIN_URI_XXX` instead.

File: sdk/keyvault/azure-security-keyvault-jca/src/main/java/com/azure/security/keyvault/jca/KeyVaultClient.java [hunk 2]
```diff
@@ -96,6 +110,28 @@
             keyVaultUri = keyVaultUri + "/";
         }
         this.keyVaultUrl = keyVaultUri;
+        String dnsSuffix = Optional.of(keyVaultUri)
```
Comment (chenrujun) [inline sdk/keyvault/azure-security-keyvault-jca/src/main/java/com/azure/security/keyvault/jca/KeyVaultClient.java:113]: `dns` is short for `Domain Name System`. So I prefer `domainNameSuffix` than `dnsSuffix`.

File: sdk/keyvault/azure-security-keyvault-jca/src/main/java/com/azure/security/keyvault/jca/KeyVaultClient.java [hunk 3]
```diff
@@ -96,6 +110,28 @@
...
+                break;
+            case KEY_VAULT_BASE_URI_US :
+                this.aadAuthenticationUrl = AAD_LOGIN_US_URI;
+                break;
+            case KEY_VAULT_BASE_URI_DE:
+                this.aadAuthenticationUrl = AAD_LOGIN_DE_URI;
+                break;
+            default:
+                throw new IllegalArgumentException("Property of azure.keyvault.uri is illegal.");
+        }
```
Comment (chenrujun) [inline sdk/keyvault/azure-security-keyvault-jca/src/main/java/com/azure/security/keyvault/jca/KeyVaultClient.java:134]: 1. Write a method in `Constants`: `getAADLoginURIByKeyVaultBaseUri`. 
2. And rename `Constants` to `UriUtil`.
3. Write unit test for the `getAADLoginURIByKeyVaultBaseUri`.

**Review by chenrujun — CHANGES_REQUESTED** [review]

File: sdk/keyvault/azure-security-keyvault-jca/src/main/java/com/azure/security/keyvault/jca/KeyVaultClient.java (hunk 3 above)
Comment (yiliuTo) [reply sdk/keyvault/azure-security-keyvault-jca/src/main/java/com/azure/security/keyvault/jca/KeyVaultClient.java:134]: Done with point 1&2. 
Do we still need a unit test for getAADLoginURIByKeyVaultBaseUri given it's just a switch statement? Does the current ut in KeyVaultClientTest meet the requirements?

**Review by backwind1233 — APPROVED** [review]
LGTM

File: sdk/keyvault/azure-security-keyvault-jca/src/test/java/com/azure/security/keyvault/jca/KeyVaultKeyManagerTest.java [hunk 4]
```diff
@@ -28,7 +28,6 @@ public void setEnvironmentProperty() throws KeyStoreException, NoSuchAlgorithmEx
         KeyStore keyStore = KeyStore.getInstance("AzureKeyVault");
         KeyVaultLoadStoreParameter parameter = new KeyVaultLoadStoreParameter(
             System.getProperty("azure.keyvault.uri"),
-            System.getProperty("azure.keyvault.aad-authentication-url"),
```
Comment (chenrujun) [inline sdk/keyvault/azure-security-keyvault-jca/src/test/java/com/azure/security/keyvault/jca/KeyVaultKeyManagerTest.java:31]: It's a breaking change, please update changelog.

**Review by chenrujun — CHANGES_REQUESTED** [review]

**Review by chenrujun — APPROVED** [review]
LGTM.

**Pull request merged**